src/vs/workbench/services/workspaces/common/workspaceTrust.ts

```diff
@@ -340,6 +335,121 @@ export class WorkspaceTrustManagementService extends Disposable implements IWork
+	) {
+		super(configurationService, storageService, uriIdentityService, environmentService, workspaceService, () => this.calculateWorkspaceTrust());
+
+		this._trustState.isTrusted = false;
+
+		// Remote - resolve remote authority
+		if (this.environmentService.remoteAuthority) {
+			this.remoteAuthorityResolverService.resolveAuthority(this.environmentService.remoteAuthority)
+				.then(async result => {
+					this._remoteAuthority = result;
+					await this.updateWorkspaceTrust();
+
+					// Show workspace trust modal dialog
+					this._onDidInitiateWorkspaceTrustRequestOnStartup.fire();
+				});
+		}
+	}
+
+	override canSetWorkspaceTrust(): boolean {
+		// Remote - remote authority not yet resolved, or remote authority explicitly sets workspace trust
+		if (this.environmentService.remoteAuthority && (!this._remoteAuthority || this._remoteAuthority.options?.isTrusted !== undefined)) {
+			return false;
+		}
+
+		return super.canSetWorkspaceTrust();
+	}
+
+	protected async calculateWorkspaceTrust(): Promise<boolean> {
+		const isTrustedDefault = this.calculateWorkspaceTrustDefault();
+		if (isTrustedDefault !== undefined) {
+			return isTrustedDefault;
+		}
+
+		// Remote - remote authority explicitly sets workspace trust
+		if (this.environmentService.remoteAuthority && this._remoteAuthority.options?.isTrusted !== undefined) {
+			return this._remoteAuthority.options.isTrusted;
+		}
+
+		// Workspace
+		const workspaceUris = await this.getWorkspaceUris();
+		const trusted = this.getUrisTrust(workspaceUris);
+
+		return trusted;
+	}
+
+	protected async getWorkspaceUris(): Promise<URI[]> {
+		const workspaceUris: URI[] = [];
+
+		// Workspace folders
+		for (const { uri } of this.workspaceService.getWorkspace().folders) {
+			workspaceUris.push(await this.getCanonicalUri(uri));
+		}
+
+		// Workspace file
+		const workspaceConfiguration = this.workspaceService.getWorkspace().configuration;
+		if (workspaceConfiguration && !isUntitledWorkspace(workspaceConfiguration, this.environmentService)) {
+			workspaceUris.push(await this.getCanonicalUri(workspaceConfiguration));
+		}
+
+		return workspaceUris;
+	}
+
+	private async getCanonicalUri(uri: URI): Promise<URI> {
+		return (this.environmentService.remoteAuthority && this._remoteAuthority) ?
+			await this.remoteAuthorityResolverService.getCanonicalURI(uri) : uri;
+	}
+}
+
+export class WorkspaceTrustManagementService extends AbstractWorkspaceTrustManagementService {
+	constructor(
+		@IConfigurationService configurationService: IConfigurationService,
+		@IStorageService storageService: IStorageService,
+		@IUriIdentityService uriIdentityService: IUriIdentityService,
+		@IWorkbenchEnvironmentService environmentService: IWorkbenchEnvironmentService,
+		@IWorkspaceContextService workspaceService: IWorkspaceContextService
+	) {
+		super(configurationService, storageService, uriIdentityService, environmentService, workspaceService, () => this.calculateWorkspaceTrust());
```

Comment:
why do we have to pass this in instead of overriding?

Reply:
That was left behind from some refactoring. Removed.